block_util/src/lib.rs

```diff
@@ -335,6 +336,12 @@ impl Request {
         // Queue operations expected to be submitted.
         match request_type {
             RequestType::In => {
+                for (data_addr, data_len) in &self.data_descriptors {
+                    mem.get_slice(*data_addr, *data_len as usize)
+                        .map_err(ExecuteError::GetHostAddress)?
+                        .bitmap()
+                        .mark_dirty(0, *data_len as usize);
+                }
```

Comment:
Nice :)
Do you have other places in mind that would require to manually mark some pages through the codebase? Just wondering if you expect to send some follow up PR or if we're good now?

Reply:
I don't have other places that need similar changes for now. I only screened the usage of `VolatileSlice::as_ptr()` in our code base. We will need to revisit this issue later when the list of these special functions is "finalized" in the upstream vm-memory crate (which is still under discussion now).

Reply:
Sounds good